packages/webamp/js/components/FFTNullsoft.ts

```diff
@@ -0,0 +1,243 @@
+// The Web Audio API's FFT is bad, so this exists now!
+// Taken from https://github.com/WACUP/vis_classic/tree/master/FFTNullsoft
+
+export class FFT {
+    private m_samples_in: number;
+    private NFREQ: number;
+    private bitrevtable: number[] | null = null;
+    private envelope: Float32Array | null = null;
+    private equalize: Float32Array | null = null;
+    private temp1: Float32Array | null = null;
+    private temp2: Float32Array | null = null;
+    private cossintable: Float32Array[] | null = null;
+
+    constructor() {
+        this.m_samples_in = 0;
+        this.NFREQ = 0;
+    }
+
+    public init(samples_in: number, samples_out: number, bEqualize = 1, envelope_power = 1.0, mode = false): void {
+        this.cleanUp();
+
+        this.m_samples_in = samples_in;
+        this.NFREQ = samples_out * 2;
+
+        this.initBitRevTable();
+        this.initCosSinTable();
+
+        if (envelope_power > 0) {
+            this.initEnvelopeTable(envelope_power);
+        }
+
+        if (bEqualize) {
+            this.initEqualizeTable(mode);
+        }
+
+        this.temp1 = new Float32Array(this.NFREQ);
+        this.temp2 = new Float32Array(this.NFREQ);
+    }
+
+    public cleanUp(): void {
+        this.envelope = null;
+        this.equalize = null;
+        this.bitrevtable = null;
+        this.cossintable = null;
+        this.temp1 = null;
+        this.temp2 = null;
+    }
```

Comment:
I don't think this ever gets called except in `init` where it has no effect because the values are already set to these values. Should we delete this?

Reply:
I've removed the cleanUp function and so far nothing bad has happened (yet)